pkg/network/ovn_kubernetes.go

```diff
@@ -532,10 +548,23 @@ func renderOVNKubernetes(conf *operv1.NetworkSpec, bootstrapResult *bootstrap.Bo
 func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootstrap.BootstrapResult) (renderIPsecMachineConfig, renderIPsecDaemonSet,
 	renderIPsecOVN, renderIPsecHostDaemonSet, renderIPsecContainerizedDaemonSet, renderIPsecDaemonSetAsCreateWaitOnly bool) {
 	isHypershiftHostedCluster := bootstrapResult.Infra.HostedControlPlane != nil
-	isIPsecEnabled := conf.IPsecConfig != nil
 	isIpsecUpgrade := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.LegacyIPsecUpgrade
 	isOVNIPsecActive := bootstrapResult.OVN.IPsecUpdateStatus != nil && bootstrapResult.OVN.IPsecUpdateStatus.OVNIPsecActive
 
+	// Find the IPsec mode from Ipsec.config
+	// Ipsec.config == nil (bw compatibility) || ipsecConfig == Off ==> ipsec is disabled
```

Comment:
I am curious on this however. I wouldn't expect the possibility of ipsecConfig==nil. Has that really been observed to happen with thew new API in place? Expecting it does not do any harm, I am just curious about it.

Reply:
This is for upgrade scenarios for a cluster which did not have ipsec enabled. It can not happen with the new API but it happens in upgrade situations.